vanilla/vanilla
# Add server rendering of Quill mentions

Closes https://github.com/vanilla/vanilla/issues/7019

- Adds server rendering for mentions.
  - Data attributes of the username and userID have been added for  https://github.com/vanilla/rich-editor/issues/65
  - The same mention url formatting has been used as our current mentions https://github.com/vanilla/vanilla/blob/feature/server-render-mention/library/core/class.format.php#L1930
- Tweaks the rendered style to not have `user-select:all` (partial selection is allowed on rendered mentions).
- Adds sanitization tests for mentions (Added one for emojis as well. They needed the same fix).
- Changes the sanization test namespace to accomodate other types of embeds.

## To Test

1. Create a post/comment with some mentions using the rich editor.
2. Observe the rendered output.
3. Test that the links go where they are supposed to.

## Known issues

The rendered output for emojis and mentions puts an extra newline after them. This will be resolved with https://github.com/vanilla/vanilla/issues/7194 which is in my current sprint.
